src/go/k8s/pkg/console/configmap.go

```diff
@@ -0,0 +1,360 @@
+
+	// Create new ConfigMap instead of updating existing so Deployment will trigger a reconcile
+	immutable := true
+	obj := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: cm.consoleobj.GetName() + "-",
+			Namespace:    cm.consoleobj.GetNamespace(),
+			Labels:       labels.ForConsole(cm.consoleobj),
+		},
+		Data: map[string]string{
+			"config.yaml": config,
+		},
+		Immutable: &immutable,
+	}
+
+	if err := controllerutil.SetControllerReference(cm.consoleobj, obj, cm.scheme); err != nil {
+		return err
+	}
+	if err := cm.Create(ctx, obj); err != nil {
+		return fmt.Errorf("creating Console configmap: %w", err)
+	}
+
+	// This will get updated in the controller main reconcile function
+	// Other Resources may set Console status if they are also watching GenerationMatchesObserved()
+	cm.consoleobj.Status.ConfigMapRef = &corev1.ObjectReference{Namespace: obj.GetNamespace(), Name: obj.GetName()}
+
+	return nil
+}
+
+// Key implements Resource interface
+func (cm *ConfigMap) Key() types.NamespacedName {
+	return types.NamespacedName{Name: cm.consoleobj.GetName(), Namespace: cm.consoleobj.GetNamespace()}
+}
+
+// generateConsoleConfig returns the actual config passed to Console.
+// This should match the fields at https://github.com/redpanda-data/console/blob/master/docs/config/console.yaml
+// We are copying the fields instead of importing them because (1) they don't have json tags (2) some fields aren't ideal for K8s (e.g. TLS certs shouldn't be file paths but Secret reference)
+func (cm *ConfigMap) generateConsoleConfig(
+	ctx context.Context, username, password string,
+) (string, error) {
+	consoleConfig := &ConsoleConfig{
+		MetricsNamespace: cm.consoleobj.Spec.MetricsPrefix,
+		ServeFrontend:    cm.consoleobj.Spec.ServeFrontend,
+		Server:           cm.genServer(),
+		Kafka:            cm.genKafka(username, password),
+	}
+
+	connectConfig, err := cm.genConnect(ctx)
+	if err != nil {
+		return "", err
+	}
+	consoleConfig.Connect = *connectConfig
+
+	config, err := yaml.Marshal(consoleConfig)
+	if err != nil {
+		return "", err
+	}
+	return string(config), nil
+}
+
+func (cm *ConfigMap) genServer() rest.Config {
+	server := cm.consoleobj.Spec.Server
+	return rest.Config{
+		ServerGracefulShutdownTimeout:   server.ServerGracefulShutdownTimeout.Duration,
+		HTTPListenAddress:               server.HTTPListenAddress,
+		HTTPListenPort:                  server.HTTPListenPort,
+		HTTPServerReadTimeout:           server.HTTPServerReadTimeout.Duration,
+		HTTPServerWriteTimeout:          server.HTTPServerWriteTimeout.Duration,
+		HTTPServerIdleTimeout:           server.HTTPServerIdleTimeout.Duration,
+		CompressionLevel:                server.CompressionLevel,
+		BasePath:                        server.BasePath,
+		SetBasePathFromXForwardedPrefix: server.SetBasePathFromXForwardedPrefix,
+		StripPrefix:                     server.StripPrefix,
+	}
+}
+
+var (
+	// Currently issuing TLS certs through LetsEncrypt
```

Comment:
I am not sure we can state this in a generally available operator when ppl can inject any cert they want, right? 🤔

Reply:
I think ideally this should be checked if `IssuerRef` is set since it identifies the cert-manager Issuer used to issue certificate. But this isn't the case in v1 - `NodeSecretRef` is set but the certificates are signed by LetsEncrypt. Is there a way to get this dynamically? :thinking_face: